indicator-sound-switcher: init at 2.3.6

Alexnortung opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

This package is added for a packaging request #136303

There are also some -wrapped files in the bin/ folder, but I don't know if they should be there.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Alexnortung committed:
indicator-sound-switcher: init at 2.3.6

This package is added for a packaging request #136303

SuperSandro2000 commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,65 @@
+  ];
+
+  makeWrapperArgs = [
+    "--prefix LD_LIBRARY_PATH : ${libpulseaudio}/lib"
```
Is this really required? Could we patch the path instead?

Alexnortung replied:
What does this mean? :)
This works, and I am unsure how to do it in another way.

bobvanderlinden replied:
I think it may be possible to instead of letting the python library reference to an arbitrary libpulse.so.0, instead explicitly reference to `${libpulseaudio}/lib/libpulse.so`.

This avoids LD_LIBRARY_PATH to unintentionally propagate to other processes if indicator-sound-switcher executes other programs.

Possibly byI looked at the project and it might be possible to 'patch' lines like:

https://github.com/yktoo/indicator-sound-switcher/blob/e0a159b349580da21809fe6b9b0a221c47317bf4/lib/indicator_sound_switcher/lib_pulseaudio.py#L7

The CDLL call loads the library, so that argument could be changed.

Patching could be done using `substituteInPlace` in `postPatch`.

Alexnortung replied:
@bobvanderlinden Thanks a lot, I understand it much better and I have made the change now :smile:

SuperSandro2000 commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,65 @@
+, gdk-pixbuf
+}:
+
+let
+  name = "indicator-sound-switcher";
+  version = "2.3.6";
+in
+python3Packages.buildPythonApplication {
+#python3Packages.buildPythonPackage {
+  pname = name;
+  inherit version;
```
```suggestion
python3Packages.buildPythonApplication rec {
  pname = "indicator-sound-switcher";
  version = "2.3.6";
```

Alexnortung replied:
Do we really need to use `rec`?

legendofmiracles replied:
Yes, because you're referencing the version attribute in the fetchFromGitHub one.

SuperSandro2000 commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,65 @@
+#python3Packages.buildPythonPackage {
+  pname = name;
+  inherit version;
+  src = fetchurl {
```
Please use fetchFromGitHub.

Alexnortung committed:
indicator-sound-switcher: etc

Requested changes

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/how-to-patch-correctly-in-my-pull-request/16521/1

legendofmiracles commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,64 @@
+, gdk-pixbuf
+}:
+
+let
+in
```
```suggestion
```

Alexnortung committed:
indicator-sound-switcher: etc

removed redundant let in

Co-authored-by: legendofmiracles <30902201+legendofmiracles@users.noreply.github.com>

Alexnortung commented:
@legendofmiracles I know I am referencing the version attribute. The point is that I wanted to avoid using `rec` and instead use `let in`. This is mostly by principle since `rec` can often lead to infinite recursion and that it is an anti pattern according to https://nix.dev/anti-patterns/language#rec-expression

legendofmiracles commented:
The issue is that rec is the standard in nixpkgs, and doing something else because you dislike it is kind of counterproductive. Consistency > something slightly better

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/how-to-patch-correctly-in-my-pull-request/16521/3

Alexnortung committed:
indicator-sound-switcher: etc

Using substituteInPlace instead of prefixing libpulseaudio to the LD_LIBRARY_PATH

Alexnortung commented:
I think this package is ready :)

SuperSandro2000 commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,64 @@
+
+python3Packages.buildPythonApplication rec {
+  pname = "indicator-sound-switcher";
+  version = "2.3.6";
+  src = fetchFromGitHub {
```
```suggestion
  version = "2.3.6";

  src = fetchFromGitHub {
```

SuperSandro2000 commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,64 @@
+    description = "Sound input/output selector indicator for Linux";
+    homepage = "https://yktoo.com/en/software/sound-switcher-indicator/";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [
+      alexnortung
+    ];
```
```suggestion
    maintainers = with maintainers; [ alexnortung ];
```

Alexnortung committed:
indicator-sound-switcher: etc

Formatting

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

Alexnortung committed:
indicator-sound-switcher: etc

Formatting

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

SuperSandro2000 commented on pkgs/applications/audio/indicator-sound-switcher/default.nix:
```diff
@@ -0,0 +1,63 @@
+  };
+
+  postPatch = ''
+    ls -lah lib
```
```suggestion
```

SuperSandro2000 committed:
Update pkgs/applications/audio/indicator-sound-switcher/default.nix
